python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -141,7 +142,12 @@ def write_partition(
 
 
 def read_parquet(
-    path, columns=None, split_row_groups=True, gather_statistics=None, **kwargs
+    path,
+    columns=None,
+    chunksize=None,
+    split_row_groups=True,
```

Comment:
Shouldn't this default to `False`?

Reply:
Sorry - I'll have to envestigate the exact cause of #3695 a bit more carefully and report back.

The upstream default is `split_row_groups=True` - and this was the case before the performance regression.  I do not think we want to change this.

It seems that the problem was most likely that we decided to set `gather_statistics=True` unless the user explicitly set `split_row_groups=False`.  Gathering statistics can be expensive in some cases.

Reply:
👍 I may be completely mistaken here, completely defer to you!

Reply:
You are not mistaken that that it can be slower to split by row-group when the corresponding row-groups are too small.  That is indeed one of the reasons for the performance regression (along with slow metadata parsing).

The repro in #3695 is reading a list of parquet files, which means the `"_metadata"` file is not being used to collect metadata.  Instead, the footer of every individual file is parsed by a single process (since we are setting `gather_statistics=True` by default).  Prior to #3407, the statistics would **not** be collected (by default) in this case, because parsing every file can be expensive on some file systems.  For this reason, the files were **not** split into row groups before the regression (because we need the metadata/file-statistics to do this).

Although it is sub-optimal to split by row-group in this particular case, I do not believe that we should make it the default behavior to split by file.  If the row groups are too small, it is up to the user to speficify `split_row_groups=False`.  With that said, we may be able to do more to encourage users to write optimally-sized row groups (defaults in the writer, etc).